src/xml_parser.h

```diff
@@ -317,6 +317,43 @@ cr_xml_parse_main_metadata_together(const char *primary_path,
                                     gboolean allow_out_of_order,
                                     GError **err);
 
+typedef struct {
+    CR_FILE *primary_f;
+    CR_FILE *filelists_f;
+    CR_FILE *other_f;
+
+    const char *primary_path;
+    const char *filelists_path;
+    const char *other_path;
+
+    void *primary_pd;
+    void *filelists_pd;
+    void *other_pd;
+
+    gboolean allow_out_of_order;
+    GError *tmp_err;
+
+    gboolean primary_is_done;
+    gboolean filelists_is_done;
+    gboolean other_is_done;
+
+    void *cbdata;
+} cr_PkgIterator;
```

Comment:
It would probably be good to hide the contents of the struct as those are implementation details.

I think doing here just `typedef struct _cr_PkgIterator cr_PkgIterator` and then have `_cr_PkgIterator` definition somewhere in private is a possible approach?

Reply:
When I try that I get a lot of "invalid use of incomplete typedef" warnings.